frontend/src/views/instructor/ddahs/ddah-editor.tsx

```diff
@@ -0,0 +1,509 @@
+import React from "react";
+import type { Ddah, Duty } from "../../../api/defs/types";
+import {
+    FaEdit,
+    FaDownload,
+    FaTrash,
+    FaPlus,
+    FaSave,
+    FaInfoCircle,
+} from "react-icons/fa";
+
+import { Alert, Button, Form, Modal, Spinner } from "react-bootstrap";
+import { formatDate, formatDownloadUrl } from "../../../libs/utils";
+import { splitDutyDescription } from "./../../../libs/ddah-utils";
+
+import "./style.css";
+import { DialogRow } from "../../../components/forms/common-controls";
+import { stringToNativeType } from "../../../libs/urls";
+
+type Category =
+    | "note"
+    | "prep"
+    | "training"
+    | "meeting"
+    | "contact"
+    | "marking"
+    | "other";
```

Comment:
Can we make this a global type? I am sure it would be useful in other files too. Additionally, we could utilize stronger type with string templates, e.g.:

```
type Category = 'meeting' | 'prep' | ... ;
type DescriptionText = string;
....
export interface RawDuty {
    order: number;
    hours: number;
    description: `${Category}:${DescriptionText}`;
};
```